src/frontend/web_application/src/components/Dropdown/index.jsx

```diff
@@ -1,145 +1,171 @@
+    children: null,
     className: null,
-    position: null,
     closeOnClick: false,
     closeOnClickExceptSelectors: null,
+    closeOnScroll: false,
+    position: 'bottom',
+    isMenu: false,
+    onToggle: str => str,
     show: false,
-    children: undefined,
-    onToggle: null,
   };
 
-  componentDidMount() {
-    this.$dropdown = jQuery(`#${this.props.id}`);
-    // eslint-disable-next-line no-new
-    new Foundation.Dropdown(this.$dropdown, {
-      hOffset: 0,
-      vOffset: 0,
-    });
-    // `positionClass` option (eg. 'right bottom') is always overriden
-    // use css classes instead (cf. alignement and position)
-    // see https://github.com/zurb/foundation-sites/pull/9019
+  state = {
+    isOpen: false,
+    offset: {},
+  };
 
-    if (this.onToggle) {
-      this.$dropdown.on('show.zf.dropdown', () => {
-        this.onToggle(true);
-      });
-      this.$dropdown.on('hide.zf.dropdown', () => {
-        this.onToggle(false);
-      });
-    }
+  componentDidMount() {
+    this.dropdownControl = document.getElementById(`${CONTROL_PREFIX}-${this.props.id}`);
 
-    if (this.props.closeOnClickExceptSelectors) {
-      this.handleDocumentClick = (ev) => {
-        if (!this.props.show) {
-          return;
-        }
+    this.handleDocumentScroll = () => { this.toggle(false); };
 
-        const target = ev.target;
+    this.handleDocumentClick = (ev) => {
+      const target = ev.target;
+      const exceptSelectors = this.props.closeOnClickExceptSelectors;
 
-        if (this.$dropdown.is(target) || this.$dropdown.find(target).length) {
-          return;
-        }
+      const dropdownClick = !this.props.closeOnClick &&
+        (this.dropdown === target || this.dropdown.contains(target));
 
+      const exeptSelectorsClick = exceptSelectors &&
+        exceptSelectors.find(selector => (selector === target));
 
-        const targetToIgnore = this.props.closeOnClickExceptSelectors
-          .find(selector => $(selector).is(target));
+      const controlClick = this.dropdownControl &&
+        (this.dropdownControl === target || this.dropdownControl.contains(target));
 
-        if (targetToIgnore) {
-          return;
-        }
+      if (controlClick) {
+        this.toggle(!this.state.isOpen);
 
-        this.show(false);
-      };
+        return;
+      }
+      if (dropdownClick) { return; }
+      if (exeptSelectorsClick) { return; }
 
-      document.addEventListener('click', this.handleDocumentClick);
-    }
+      this.toggle(false);
+    };
 
-    this.show(this.props.show);
+    this.toggle(this.props.show);
+    document.addEventListener('click', this.handleDocumentClick);
+    if (this.props.closeOnScroll) { window.addEventListener('scroll', this.handleDocumentScroll); }
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.show !== nextProps.show) {
-      this.show(nextProps.show);
-    }
+    if (this.props.show !== nextProps.show) { this.toggle(nextProps.show); }
```

Comment:
_(my preference goes for multiple lines for conditions, that I find more readable. Also you might want to drop braces to keep one-line condition, but I still find this less readable than multiple lines + braces)_

_(less lines of code is not better, less code is :) )_

Do you really prefer one-line condition ? We can set it in eslint.